modules/imap/output_modules.php

```diff
@@ -101,17 +102,32 @@ protected function output() {
             }
             if (isset($struct['subtype']) && strtolower($struct['subtype']) == 'html') {
                 $allowed = $this->get('header_allow_images');
```

Comment:
If we stop using header_allow_images and imap_allow_images, shouldn't we get rid of them in the handler methods as well? Also what is going to happen with the allow_external_image_sources config setting? If we don't respect it, then we should probably remove it from settings.

Reply:
OK, I made some more investigations and think we should do:
- respect allow_external_image_sources application config setting - if that's off - Content-securty-policy disables external resource loading altogether, so it doesn't makes sense to do any of this data-src replacing and handling - it is probably best to just keep the old URI.DisableExternalResources setting or replace the src attribute and skip adding data-src. I think keep the old setting is more robust as we might have missed a use-case of the src matching.
- if that's on, we go ahead and do the replacing of data-src and other logic.
All other variables like imap_allow_images should be removed from handlers as they are not used.

Reply:
OK, then we'll have also to not display the warning messages to the user about external resources in case the setting allow_external_image_sources was off.